test/metabase/dashboard_subscription_test.clj

```diff
@@ -881,7 +885,55 @@
                   [{:type "divider"}
                    {:type "context",
                     :elements
-                    [{:type "mrkdwn",
+                    [{:type "mrkdwn"
                       :text
                       #"<https://metabase\.com/testmb/dashboard/\d+\?state=CA&state=NY&state=NJ&quarter_and_year=Q1-2021\|\*Sent from Metabase Test\*>"}]}]}]}
                (pulse.test-util/thunk->boolean pulse-results))))}}))
+
+(defn- result-attachment
+  [{{{:keys [rows]} :data, :as result} :result}]
+  (when (seq rows)
+    [(let [^java.io.ByteArrayOutputStream baos (java.io.ByteArrayOutputStream.)]
+       (with-open [os baos]
+         (#'messages/stream-api-results-to-export-format :csv os result)
+         (let [output-string (.toString baos "UTF-8")]
+           {:type         :attachment
+            :content-type :csv
+            :content      output-string})))]))
+
+(defn- metadata->field-ref
+  [{:keys [name field_ref]} enabled?]
+  {:name name :field_ref field_ref :enabled enabled?})
+
+(deftest dashboard-subscription-attachments-test
+  (testing "Dashboard subscription attachments respect dashcard viz settings."
+    (mt/with-fake-inbox
+      (mt/with-temp [Card {card-id :id :as c} (pulse.test-util/checkins-query-card {:breakout [!day.date]})
+                     Dashboard     {dash-id :id} {:name "just dash"}]
+        (let [viz {:table.columns (mapv metadata->field-ref (:result_metadata c) [true false])}]
```

Comment:
Might be worth a comment as to what these viz settings do so the reader knows what the expected applied changes are.

Reply:
Good call, added.